src/Repository/PetRepository.php

```diff
@@ -21,38 +22,35 @@ public function __construct(RegistryInterface $registry)
         parent::__construct($registry, Pet::class);
     }
 
-    public function searchPets(Pet $pet)
+    public function searchPets(PetDTO $petDTO)
     {
         $queryBuilder = $this->getEntityManager()->createQueryBuilder();
+        $parameters = [];
         $queryBuilder->select('p')
             ->from('App:Pet', 'p');
-        if ($pet->getSpecies() !== null) {
-            $queryBuilder->where($queryBuilder->expr()->eq(
-                'p.species',
-                "'".$pet->getSpecies()->getId()."'"
-            ));
+
+        if ($petDTO->getSpecies() !== null) {
+            $parameters['species'] = $petDTO->getSpecies()->getId();
+            $queryBuilder->where('p.species = :species');
```

Comment:
there should be `setParameter` IIRC, and you can set it immediately, instead of storing values in a new array.